src/main/java/core/basesyntax/service/impl/FruitListImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.FruitList;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitListImpl implements FruitList {
+    @Override
+    public List<FruitTransaction> getFruitList(List<FruitTransaction> list) {
+        List<String> listOfFruitsString = list.stream()
+                .map(FruitTransaction::getFruit)
+                .distinct()
+                .toList();
+        List<FruitTransaction> listOfFruitsAsObjects = new ArrayList<>();
+        for (String fruit : listOfFruitsString) {
+            int defaultQuantity = 0;
+            listOfFruitsAsObjects.add(new FruitTransaction(fruit, defaultQuantity));
+        }
```

Comment:
Using a for-each loop to create 'FruitTransaction' objects with a default quantity of 0 is not efficient. Consider simplifying this logic by using a stream operation to map the distinct fruit names directly to 'FruitTransaction' objects.